### PR TITLE
Data Types Testing Issue

### PR DESCRIPTION
**Pull request by khalilTN**
The tests of data types (semantic/types) use the values returned from a .create() call, .create() retrieved only auto incremented ids from the table so the values returned in callback will be the same values passed as parameters, that's why the test should insert, and then query the table to retrieve the persistent value.

### CODE REVIEW

**Comment by devinivy** [comment]
Looks good– thanks! :+1:

**Comment by khalilTN** [comment]
You are welcome ! ;)

**Comment by dmarcelino** [comment]
@khalilTN, @atiertant, this does look good indeed! This has exposed an apparent bug in sails-mongo since it is breaking the test:
* Semantic Interface Binary Type with valid data should store proper binary value

build log line 536: https://travis-ci.org/balderdashy/waterline-adapter-tests/jobs/61589255#L536

We'll need to fix that before we can merge this to avoid breaking the build. I've raised balderdashy/sails-mongo#272 to address the sails-mongo issue.

**Comment by atiertant** [comment]
@dmarcelino there is no better demonstation about this tests are useful...

**Comment by dmarcelino** [comment]
Indeed! :smiley:

@khalilTN, @atiertant, it just occurred to me that this change make us lose the verification of the values returned by `.create()`. Can we also add checks to those, like:
```javascript
  assert(createdRecord.list.length === 4);
```

File: interfaces/semantic/types/array.type.js [hunk 1]
```diff
@@ -11,14 +11,16 @@ describe('Semantic Interface', function() {
       ////////////////////////////////////////////////////
 
       it('should store proper array value', function(done) {
-        Semantic.User.create({ list: [0,1,2,3] }, function(err, record) {
+        Semantic.User.create({ list: [0,1,2,3] }, function(err, createdRecord) {
           assert(!err);
-          assert(Array.isArray(record.list));
-          assert(record.list.length === 4);
```
Comment (dmarcelino) [inline interfaces/semantic/types/array.type.js]: We should still check if the `createdRecord` returns the right value, like: `assert(createdRecord.list.length === 4);` for all types.

**Comment by dmarcelino** [comment]
I've submitted PR balderdashy/sails-mongo#274 which fixes balderdashy/sails-mongo#272.

**Comment by khalilTN** [comment]
I have commited the tests for the record returned by .create().

**Comment by dmarcelino** [comment]
Perfect @khalilTN! Thank you.

**Pull request merged**